src/libtmux/session.py

```diff
@@ -259,6 +260,10 @@ def new_window(
             % (self.id, window_index),
         )
 
+        if environment:
+            for k, v in environment.items():
+                window_args += ("-e%s=%s" % (k, v),)
```

Comment:
tmux 3.0 compatible: https://github.com/tmux/tmux/blob/42358cc/CHANGES#L823

Reply:
Will take a look.

Reply:
You can take another look. If you're okay with the changes I'd fixup the commits and push them again.

Reply:
Looks good to me (to fix up)